base/views.py

```diff
@@ -28,8 +48,20 @@ def signup(request):
         
         try:
             user = User.objects.create_user(username=email, email=email, password=password, cnic=cnic, dob=dob, phone=phone, address=address)
+            
```

Comment:
create the token before User object creation and, then pass the token in `create_user` to avoid ` user.verification_token = token
            user.save()`